spotinst/resource_spotinst_credentials_gcp.go

```diff
@@ -0,0 +1,105 @@
+package spotinst
+
+import (
+	"context"
+	"fmt"
+	"github.com/spotinst/spotinst-sdk-go/service/account/providers/gcp"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/credentials_gcp"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/client"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+)
+
+func resourceSpotinstCredentialsGCP() *schema.Resource {
+	setupCredentialsGCPResource()
+
+	return &schema.Resource{
+		CreateContext: resourceSpotinstCredentialsGCPCreate,
+		ReadContext:   resourceSpotinstCredentialsGCPRead,
+		DeleteContext: schema.NoopContext,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: commons.CredentialsGCPResource.GetSchemaMap(),
+	}
+}
+
+func setupCredentialsGCPResource() {
+	fieldsMap := make(map[commons.FieldName]*commons.GenericField)
+
+	credentials_gcp.Setup(fieldsMap)
+
+	commons.CredentialsGCPResource = commons.NewCredentialsGCPResource(fieldsMap)
+}
+
+func resourceSpotinstCredentialsGCPCreate(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	log.Printf(string(commons.ResourceOnCreate),
+		commons.CredentialsGCPResource.GetName())
+
+	credentials, err := commons.CredentialsGCPResource.OnCreate(resourceData, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	resourceData.SetId(spotinst.StringValue(credentials.AccountId))
+
+	err = createGCPCredentials(credentials, meta.(*Client))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("===> Account created successfully: %s <===", resourceData.Id())
+	return resourceSpotinstCredentialsGCPRead(ctx, resourceData, meta)
+
+}
+
+func createGCPCredentials(credentials *gcp.ServiceAccounts, spotinstClient *Client) error {
+	if json, err := commons.ToJson(credentials); err != nil {
+		return err
+	} else {
+		log.Printf("===> Credentials configuration: %s", json)
+	}
+	input := &gcp.SetServiceAccountsInput{ServiceAccounts: credentials}
+	_, err := spotinstClient.account.CloudProviderGCP().SetServiceAccount(context.Background(), input)
+	if err != nil {
+		return fmt.Errorf("[ERROR] failed to set credential: %s", err)
+	}
+	return nil
+}
+
+func resourceSpotinstCredentialsGCPRead(ctx context.Context, resourceData *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	id := resourceData.Id()
+	log.Printf(string(commons.ResourceOnRead),
+		commons.CredentialsGCPResource.GetName(), id)
+	input := &gcp.ReadServiceAccountsInput{AccountId: spotinst.String(id)}
+	resp, err := meta.(*Client).account.CloudProviderGCP().ReadServiceAccount(context.Background(), input)
+	if err != nil {
+		// If the account was not found, return nil so that we can show
```

Comment:
please modify the message to read as service account

Reply:
Updated comment